go/checks.rst

```diff
@@ -0,0 +1,302 @@
+| `whitelist` key, the check will not apply to that file.                                          |
++----------------------------+---------------------------------------------------------------------+
+
+Example
+^^^^^^^
+
+The following configuration file configures the checks named ``mycheck``,
+``unsafedom``, and ``loopclosure``.
+
+.. code:: json
+
+    {
+      "mycheck": {
+        "severity": "ERROR",
+        "whitelist": {
+          "src/foo.go": "manually verified that behavior is working-as-intended"
+          "src/bar.go": "see issue #1337"
+        }
+      },
+      "unsafedom": {
+        "severity": "WARNING",
+        "apply_to": {
+          "src/js/*": ""
+        },
+        "whitelist": {
+          "src/(third_party|vendor)/*": "enforce DOM safety requirements only on first-party code"
+        }
+      },
+      "loopclosure": {
+        "description": "fail builds without exception since we know this check is 100% accurate",
+        "severity": "ERROR"
+      }
+    }
+
+This label referencing this configuration file must be provided as the
+``config`` attribute value of the ``go_checker`` rule.
+
+.. code:: bzl
+
+    go_checker(
+        name = "my_checker",
+        deps = [
+            ":mycheck",
+            ":unsafedom",
+            "@javascript_checks//:loopclosure",
+        ],
+        config = ":config.json"
+        visibility = ["//visibility:public"],
+    )
+
+API
+---
+
+go_checker
+~~~~~~~~~~
+
+This generates a checker program that is run alongside the compiler to analyze
+Go source code.
+
+Providers
+^^^^^^^^^
+
+* GoChecker_
+
+Attributes
+^^^^^^^^^^
+
++----------------------------+-----------------------------+---------------------------------------+
+| **Name**                   | **Type**                    | **Default value**                     |
++----------------------------+-----------------------------+---------------------------------------+
+| :param:`name`              | :type:`string`              | |mandatory|                           |
++----------------------------+-----------------------------+---------------------------------------+
+| A unique name for this rule.                                                                     |
++----------------------------+-----------------------------+---------------------------------------+
+| :param:`deps`              | :type:`label_list`          | :value:`None`                         |
++----------------------------+-----------------------------+---------------------------------------+
+| List of Go libraries that will be linked to the generated checker binary.                        |
+| These libraries must call analysis.Register to ensure that the analyses they implement are       |
+| called by the checker binary.                                                                    |
+| These libraries must be `go_tool_library` rules to avoid bootstrapping problems.                 |
```

Comment:
`go_tool_binary` should either be formatted as code or link to the rule; ideally both, but I don't think ReST can do that.

Reply:
Done.